sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogEventSuite.scala

```diff
@@ -209,4 +209,67 @@ class ExternalCatalogEventSuite extends SparkFunSuite {
     catalog.dropFunction("db5", "fn4")
     checkEvents(DropFunctionPreEvent("db5", "fn4") :: DropFunctionEvent("db5", "fn4") :: Nil)
   }
+
+  testWithCatalog("partition") { (catalog, checkEvents) =>
+    val path1 = Files.createTempDirectory("db_")
+    val path2 = Files.createTempDirectory(path1, "tbl_")
+    val uri1 = preparePath(path1)
+    val uri2 = preparePath(path2)
+
+    // CREATE
+    val dbDefinition = createDbDefinition(uri1)
+
+    val storage = CatalogStorageFormat.empty.copy(
+      locationUri = Option(uri2))
+
+    val tableDefinition = CatalogTable(
+      identifier = TableIdentifier("tbl1", Some("db5")),
+      tableType = CatalogTableType.MANAGED,
+      storage = storage,
+      schema = new StructType().add("id", "long"))
+
+    val partition = CatalogTablePartition(spec = CatalogTypes.emptyTablePartitionSpec,
+      storage = CatalogStorageFormat.empty)
+
+    catalog.createDatabase(dbDefinition, ignoreIfExists = false)
+    checkEvents(CreateDatabasePreEvent("db5") :: CreateDatabaseEvent("db5") :: Nil)
+
+    catalog.createTable(tableDefinition, ignoreIfExists = false)
+    checkEvents(CreateTablePreEvent("db5", "tbl1") :: CreateTableEvent("db5", "tbl1") :: Nil)
+
+    catalog.createPartitions("db5", "tbl1", Seq(partition), ignoreIfExists = false)
```

Comment:
Ur, is this a valid test case? What happens when we try to create `CatalogTypes.emptyTablePartitionSpec`?

Reply:
agree with you, `CatalogTypes.emptyTablePartitionSpec` is not good option for unit test.  I have changed it.

Reply:
Despite this, I think that empty partition creation should also produce the event.

Reply:
Does it happen in the real world? Can you give me an SQL example to create the empty partition?
> Despite this, I think that empty partition creation should also produce the event.

Reply:
hm, I cannot find an example, that's why I also think it isn't happening in the real world.